appsmithorg/appsmith
# Return hint on localhost url

## Description
- Return hint for localhost URL on the following events: 
  - Test datasource
  - Save datasource
  - Update REST API Url
  - Create datasource from REST API
  - When page gets loaded for first time. 
- Hint message returned: "You may not able to access your localhost if Appsmith is running inside a docker container or on the cloud. Please check out Appsmith's documentation to understand more."
- It has been decided as part of actions pod meeting that these messages will not be persisted. 
  
## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

**Test datasource response:**
———————— 
`{"responseMeta":{"status":200,"success":true},"data":{"invalids":["Missing username for authentication."], "messages":["You may not able to access localhost if Appsmith is running inside a docker container"], "success":false}}`

**Save datasource response:** 
———————— 
`{"responseMeta":{"status":200,"success":true},"data":{"id":"6052f64a0f724760d3e94a23","userPermissions":["execute:datasources","manage:datasources","read:datasources"],"name":"Untitled Datasource 5467","pluginId":"5c9f512f96c1a50004819786","organizationId":"5f9fc0550b682b2a63e424d3","datasourceConfiguration":{"connection":{"mode":"READ_WRITE"},"endpoints":[{"host":"localhost"}],"authentication":{"authenticationType":"dbAuth","authenticationType":"dbAuth","databaseName":"admin"},"sshProxyEnabled":false},"invalids":["Missing username for authentication."],"messages":["You may not able to access localhost if Appsmith is running inside a docker container"]`,"isValid":false,"new":false}}`
`
**API URL update response:**
——————————— 
`{"responseMeta":{"status":200,"success":true},"data":{"id":"60530960bf7f4b4dceae17aa","organizationId":"5f9fc0550b682b2a63e424d3","pluginType":"API","pluginId":"5ca385dc81b37f0004b4db85","name":"Api6","datasource":{"userPermissions":[],"name":"DEFAULT_REST_DATASOURCE","pluginId":"5ca385dc81b37f0004b4db85","organizationId":"5f9fc0550b682b2a63e424d3","datasourceConfiguration":{`"url":"http://localhost"`},"invalids":[],`"messages":["You may not able to access localhost if Appsmith is running inside a docker container"]`,"isValid":true,"new":true},"pageId":"60211622f3d65469ffa7d5f5","actionConfiguration":{"timeoutInMillisecond":10000,"paginationType":"NONE","headers":[{"key":"","value":""},{"key":"","value":""}],"encodeParamsToggle":true,"queryParameters":[{"key":"","value":""},{"key":"","value":""}],"httpMethod":"GET"},"executeOnLoad":false,"dynamicBindingPathList":[],"isValid":true,"invalids":[],"jsonPathKeys":[],"confirmBeforeExecute":false,"userPermissions":["read:actions","execute:actions","manage:actions"]}}
`

**API save as datasource response:**
———————————— 
`{"responseMeta":{"status":201,"success":true},"data":{"id":"605309f5bf7f4b4dceae17ab","userPermissions":["execute:datasources","manage:datasources","read:datasources"],"name":"Untitled Datasource 5468","pluginId":"5ca385dc81b37f0004b4db85","organizationId":"5f9fc0550b682b2a63e424d3","datasourceConfiguration":{"properties":[{"key":"isSendSessionEnabled","value":"N"},{"key":"sessionSignatureKey"}],`"url":"http://localhost"`},"invalids":[],`"messages":["You may not able to access localhost if Appsmith is running inside a docker container"],`"isValid":true,"new":false}}`

**Fetch datasource info on page load response:**
————————————— 
`{"id":"6051d88334f958353c5ae719","userPermissions":["execute:datasources","manage:datasources","read:datasources"],"name":"Untitled Datasource 5460","pluginId":"5ca385dc81b37f0004b4db85","organizationId":"5f9fc0550b682b2a63e424d3","datasourceConfiguration":{"authentication":{"authenticationType":"oAuth2","grantType":"client_credentials","isTokenHeader":true,"isAuthorizationHeader":false,"headerPrefix":"Bearer "},"sshProxyEnabled":false,"properties":[{"key":"isSendSessionEnabled","value":"N"},{"key":"sessionSignatureKey","value":""}],`"url":"https://localhost",`"headers":[]},"invalids":["Missing Client Secret","Missing Client ID","Missing Access Token URL"],`"messages":["You may not able to access localhost if Appsmith is running inside a docker container"]`,"isValid":false,"new":false},`

`{"id":"6051db4bec994c15970cc8a3","userPermissions":["execute:datasources","manage:datasources","read:datasources"],"name":"Untitled Datasource 5461","pluginId":"5ca385dc81b37f0004b4db85","organizationId":"5f9fc0550b682b2a63e424d3","datasourceConfiguration":{"authentication":{"authenticationType":"oAuth2","grantType":"client_credentials","isTokenHeader":true,"isAuthorizationHeader":false,"headerPrefix":"Bearer "},"sshProxyEnabled":false,"properties":[{"key":"isSendSessionEnabled","value":"N"},{"key":"sessionSignatureKey","value":""}],`"url":"https://localhost"`,"headers":[]},"invalids":["Missing Client Secret","Missing Client ID","Missing Access Token URL"],`"messages":["You may not able to access localhost if Appsmith is running inside a docker container"]`,"isValid":false,"new":false},`

`{"id":"6051db8fec994c15970cc8a4","userPermissions":["execute:datasources","manage:datasources","read:datasources"],"name":"Untitled Datasource 5462","pluginId":"5ca385dc81b37f0004b4db85","organizationId":"5f9fc0550b682b2a63e424d3","datasourceConfiguration":{"properties":[{"key":"isSendSessionEnabled","value":"N"},{"key":"sessionSignatureKey"}],`"url":"https://localhost"`},"invalids":[],`"messages":["You may not able to access localhost if Appsmith is running inside a docker container"]`,"isValid":true,"new":false},`


## Checklist:
